gateway-service/src/test/java/org/zowe/apiml/gateway/conformance/VerificationOnboardServiceTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package org.zowe.apiml.gateway.conformance;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.cloud.client.DefaultServiceInstance;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+
+@ExtendWith(MockitoExtension.class)
+public class VerificationOnboardServiceTest {
+
+    @InjectMocks
+    private VerificationOnboardService verificationOnboardService;
+
+    @Mock
+    private DiscoveryClient discoveryClient;
+    @Nested
+    class GivenRegisteredService {
+
+        @Test
+        void whenServiceId_Registered() {
```

Comment:
these tests can be parametrized, you can pass valid and invalid inputs and assert different results. For inspiration, take a look at LoadBalancerErrorCheckTest.